backend/helpers/auth.js

```diff
@@ -5,45 +5,57 @@ const salt = bcrypt.genSaltSync(10);
 const hash = bcrypt.hashSync("password", salt);
 
 export default (db) => {
-
-	const islanders = db.collection('islanders')
-	const islands = db.collection('islanders')
-
-	const findUserById = async (userId) => {
-		const _id = new ObjectID.createFromHexString(userId)
-
-		return await islanders.findOne({ _id });
-	};
-
-	const validateUser = async (email, password) => {
-		const user = await islanders.findOne({ email });
-		console.log(user)
-		return bcrypt
-			.compare(password, user ? user.password : null)
-			.then((result) => (result ? user : null))
-			.catch(() => null);
-	};
-
-	const createUser = async ({ name, password, islandId, email, avatarImage }) => {
-		const _id = new ObjectID.createFromHexString(islandId)
-		const hashedPassword = await bcrypt.hash(password, salt)
-		const islanderInfo = {
-			_id: new ObjectID(),
-			name,
-			password: hashedPassword,
-			email,
-			avatarImage,
-			recipes: [],
-		}
-
-		const newIslander = await islanders.insertOne(islanderInfo)
-		await islands.updateOne(
-			{ _id },
-			{ $push: { islanders: newIslander.insertedId } }
-		)
-		return islanderInfo
-	};
-
-	return { findUserById, validateUser, createUser }
-
-}
\ No newline at end of file
+    const islanders = db.collection("islanders");
+    const islands = db.collection("islanders");
+
+    const findUserById = async (userId) => {
+        const _id = new ObjectID.createFromHexString(userId);
+
+        return await islanders.findOne({ _id });
+    };
+
+    const validateUser = async (email, password) => {
+        const user = await islanders.findOne({ email });
+        console.log("User found:", user);
+        console.log("password to match", password, user.password);
+        return bcrypt
+            .compare(password, user ? user.password : null)
+            .then((isValidated) => {
+                if (isValidated) return user;
+                if (user) console.log("Bad password");

```

Comment:
Does bcrypt return the user is the password isn't valid?

Reply:
No! `isValidated` is the result of whether the password matched. L25 just checks if a user was actually found so we can log it for our own debugging needs. We might eventually remove this if statement, though.